projects/igniteui-angular/src/lib/services/theme/theme.token.ts

```diff
@@ -0,0 +1,52 @@
+import { inject, InjectionToken } from "@angular/core";
+import { mkenum } from "../../core/utils";
+import { BehaviorSubject } from "rxjs";
+import { DOCUMENT } from "@angular/common";
+
+export class ThemeToken {
+    private document = inject(DOCUMENT);
+    public subject: BehaviorSubject<IgxTheme>;
+
+    constructor(private t?: IgxTheme) {
+        const globalTheme = globalThis.window
```

Comment:
And I think I preferred this more on the root factory level or was this intentional for a sneaky auto-init on a scoped bit of the DOM? I guess that could be useful, just figured the main use would be to always set something anyway which would make the computed styles call redundant.

Reply:
The sneaky thing this allows me to do is to initialize a ThemeToken without passing a theme to the constructor in the root factory, making the preferToken getter return false when no theme is passed to the factory.